x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanStreamInput.java

```diff
@@ -206,4 +210,17 @@ static void throwOnNullOptionalRead(Class<?> type) throws IOException {
     public NameId mapNameId(long l) {
         return nameIdFunction.apply(l);
     }
+
+    public FieldAttribute attributeFromCache(int id) {
+        assert fieldAttributes.length > id && fieldAttributes[id] != null;
```

Comment:
Let's have a proper IAE here with a good error message.

nit/bikeshed: why not return null in case it's not in the cache?

Reply:
It should never happen, if I got an ID from the network, that ID should be in the cache already, otherwise there is a problem.
I think an IAE is appropriate

Reply:
I think an exception is right here. Getting null means a programming bug. You can throw an `IOException` here to signal that it's a serialization bug. IllegalArgumentExcetion gets used in our code sometimes to signal that the request over the API was invalid.